### ISSUE TITLE
Command-T - ruby based plugin not working

### ISSUE THREAD

**Issue by brt** [neovim/neovim#5332]
- `nvim --version`: NVIM 0.1.6-dev
- Vim (7.4, Included patches: 1-1782) behaves differently? YES
- Operating system/version: Debian Stretch

The [Command-T](https://github.com/wincent/command-t) file window can be opened just fine, but only for the first time... when you try opening it again, the following error is displayed (at which point Command-T no longer works). This issue does not occur in regular Vim. 

```
Error detected while processing function commandt#FileFinder:
line    2: 
W, [2016-09-12T17:08:05.929999 #11983]  WARN -- Neovim::Host: got unexpected error #<ArgumentError: Invalid window id>
Error detected while processing function commandt#FileFinder[2]..provider#ruby#Call:
line   17:
Invalid window id
Press ENTER or type command to continue
```

Steps to reproduce:
Hit `<Leader>t`, pick a file and hit `<CR>`, then hit `<Leader>t` again

System info:

> Debian Stretch, NVIM 0.1.6-dev, neovim gem 0.3.0, current master branch of command-t

See also: https://github.com/neovim/neovim/issues/5223#issuecomment-241876889 (@c0nn3r already reported this behavior in comments to other possibly related Ruby plugin issue, he didn't get back to @alexgenco though, so I figured I'd submit a new one).

**Comment by justinmk** [comment]
`CheckHealth` ?

**Comment by brt** [comment]
```
health#provider#check
================================================================================
## Ruby provider
  - SUCCESS: Found up-to-date neovim RubyGem
  - INFO: Ruby Version: ruby 2.3.1p112 (2016-04-26) [x86_64-linux-gnu]
  - INFO: Host Executable: /usr/local/bin/neovim-ruby-host
  - INFO: Host Version: 0.3.0
```

**Comment by alexgenco** [comment]
Sorry I'm out of town for the week. I'll have a look as soon as I can.

**Comment by brt** [comment]
Good to hear that you'll find the time, thank you.

**Comment by jwnx** [comment]
Found same behavior.

Operating System: Arch Linux  4.7.4-1
`nvim --version` : NVIM 0.1.5

```
## Ruby provider
  - SUCCESS: Found Neovim RubyGem
  - INFO: Ruby Version: ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-linux]
  - INFO: Host Executable: /home/drstein/.gem/ruby/2.3.0/bin/neovim-ruby-host
  - INFO: Host Version: 0.3.0
```

**Comment by justinmk** [comment]
Try latest ruby neovim gem.

**Comment by alexgenco** [comment]
I started looking into this last night, but haven't nailed down the problem yet. It probably has to do with the `$curwin` variable getting out of sync, because replacing it with `VIM::Window.current`  everywhere in command-t fixed it.

**Comment by alexgenco** [comment]
Ok I figured out how to partially fix this by reverting some window caching I was doing in the provider.

However now it fails when running command-t from a tab. I've narrowed it down to the fact that, given a new tab with `:tabe`, the `vim_get_windows` RPC call is returning 2 windows, even though `vim` returns `1` when running `:ruby puts VIM::Window.count` in the same situation.

@justinmk might this be a bug with the `vim_get_windows` call?

**Comment by justinmk** [comment]
@alexgenco Can you add a failing test case to `test/functional/api/tabpage_spec.lua` ?

**Comment by alexgenco** [comment]
@justinmk I just added it to an existing list_wins spec since it seemed more relevant there: https://github.com/neovim/neovim/pull/5371

**Comment by fsouza** [comment]
I've been facing this issue with lusty explorer as well.

**Comment by brt** [comment]
@alexgenco: thank you very much, it works now!

@fsouza: Alex fixed it [here](https://github.com/alexgenco/neovim-ruby/commit/1c2e41c56e8984d04edeb8fc61884593c1f925a5) but I think it might take a while before the change makes it to the next release... so ATM in order to use it, you need to clone [neovim ruby gem repo](https://github.com/alexgenco/neovim-ruby) and install from source ([and it's just 2 commands](http://stackoverflow.com/questions/2577346/how-to-install-gem-from-github-source/2577404#2577404)).

**Comment by fsouza** [comment]
@brt thanks, it worked!

@alexgenco thanks for fixing it :)

**Comment by alexgenco** [comment]
Good to hear, folks. Unfortunately tabs are still not working, so I'll probably wait until we can get that resolved to push the gem.

**Comment by alexgenco** [comment]
Ok tabpage issue should be fixed, pushed [0.3.1](https://rubygems.org/gems/neovim/versions/0.3.1) to rubygems.
